sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -178,7 +178,17 @@ object FunctionRegistry {
 
     // datetime functions
     expression[CurrentDate]("current_date"),
-    expression[CurrentTimestamp]("current_timestamp")
+    expression[CurrentTimestamp]("current_timestamp"),
+    expression[DateFormatClass]("date_format"),
+    expression[Year]("year"),
+    expression[Quarter]("quarter"),
+    expression[Month]("month"),
+    expression[Day]("day"),
```

Comment:
@rxin In Jira you mentioned there should be an alias. Can I just add `expression[Day]("day_of_month")`?